#!/usr/bin/env python
PACKAGE = "image_enhancement"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("dilation_size",   int_t,	  0, "dilate element size", 1,  1, 50)
gen.add("subscribe_topic",   str_t,   0, "Subscribe Topic",  "")
gen.add("publish_topic",     str_t,   0, "Publish Topic",  "")

element_forms = gen.enum([ gen.const("Rect",    int_t, 0, "Rect"),
                       gen.const("Cross",     int_t, 1, "Cross"),
                       gen.const("Ellipse",     int_t, 2, "Ellipse")],
                     "erode element forms")

gen.add("element_form", int_t, 0, "Element form selector", 0, 0, 2, edit_method=element_forms)

exit(gen.generate(PACKAGE, "image_enhancement", "dilate"))
